apps/web/src/app/auth/login/login-with-device.component.ts

```diff
@@ -60,5 +61,22 @@ export class LoginWithDeviceComponent
       stateService,
       loginService
     );
+    this.onSuccessfulLogin = this.goAfterLogIn;
+  }
+
+  async goAfterLogIn() {
+    this.loginService.clearValues();
+    let previousUrl = await this.stateService.getPreviousUrl();
+    if (previousUrl == "" || previousUrl == undefined || previousUrl == null) {
```

Comment:
You can use the helper function for this.

```suggestion
    if (!Utils.isNullOrEmpty(previousUrl)) {
```

Reply:
That's the one I was thinking of. Thanks, it felt wrong to write it that way.

Reply:
Fixed in most this [commit](https://github.com/bitwarden/clients/pull/5733/commits/a525c8066ddda02f47539f7ad1bc3be478bddc63.)